zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -269,10 +261,10 @@ pub(crate) fn sighash(
                 index: input_index,
                 script_code: &script,
                 script_pubkey: &script,
-                value: output
-                    .value
-                    .try_into()
-                    .expect("amount was previously validated"),
+                value: zp_tx::components::amount::NonNegativeAmount::from_nonnegative_i64_le_bytes(
+                    output.value.to_bytes(),
+                )
+                .expect("amount was previously validated"),
```

Comment:
Nitpick: Could we update the `TryFrom` impl above (line ~209) to:

```rust
/// Convert a Zebra non-negative Amount into a librustzcash one.
impl TryFrom<Amount<NonNegative>> for zp_tx::components::amount::NonNegativeAmount {
    type Error = ();

    fn try_from(amount: Amount<NonNegative>) -> Result<Self, Self::Error> {
        zp_tx::components::amount::NonNegativeAmount::from_nonnegative_i64(amount.into())
    }
}
```

And use `try_into()` here and on line 36?

```suggestion
                value: output
                    .value
                    .try_into()
                    .expect("amount was previously validated"),
```

Reply:
Thanks, i used this at https://github.com/ZcashFoundation/zebra/pull/8522/commits/617fc03b5780cd322de41bb273507d811b53afd0